### ISSUE TITLE
Consider honoring `PropertyName`

### ISSUE THREAD

**Issue by Goooler** [detekt/detekt#7401]
## Expected Behavior of the rule

```kt
open class Foo {

  @Suppress("PropertyName")
  protected val _foo = mutableListOf<String>()
}
```

This `_foo` name should be suppressed in IDEA but not be honored by Detekt:

```sh
Foo.kt:6:17: Variable names should match the pattern: [a-z][A-Za-z0-9]* [VariableNaming]
``` 

related rules like `PrivatePropertyName`, `ConstPropertyName` and so on should also be honored.

Detekt version: v1.23.6.

## Context

The simular issue has been filed to https://github.com/pinterest/ktlint/issues/2612.

**Comment by BraisGabin** [comment]
That should be as easy as adding a new `alias` to that/those rule/rules. PRs welcome.

**Comment by atulgpt** [comment]
We can add `PropertyName` but adding `PrivatePropertyName` and `ConstPropertyName` will create a misunderstanding as we have the same rule for all types of variables. So with alias `PrivatePropertyName`, this will disable all name checks for all the variables whether that is public, private or const

**Comment by atulgpt** [comment]
I added one PR which add `PropertyName` alias for `VariableNaming` rule
